Error is reproducible on aarch64-linux:
```
mlir> FAILED: tools/mlir-rewrite/CMakeFiles/mlir-rewrite.dir/mlir-rewrite.cpp.o
mlir> /nix/store/8p08lgzbiqvj5fni1d9hxc6ma1z0mh2z-gcc-wrapper-14-20241116/bin/g++ -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/build/mlir-src-20.1.0/mlir/build/tools/mlir-rewrite -I/build/mlir-src-20.1.0/mlir/tools/mlir-rewrite -I/build/mlir-src-20.1.0/mlir/build/include -I/build/mlir-src-20.1.0/mlir/include -I/build/mlir-src-20.1.0/mlir/tools/mlir-rewrite/../../../clang/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -Wimplicit-fallthrough -Wno-nonnull -Wno-class-memaccess -Wno-redundant-move -Wno-pessimizing-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wsuggest-override -Wno-comment -Wno-misleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -Wundef -O3 -DNDEBUG -std=gnu++17  -fno-exceptions -funwind-tables -fno-exceptions -funwind-tables -MD -MT tools/mlir-rewrite/CMakeFiles/mlir-rewrite.dir/mlir-rewrite.cpp.o -MF tools/mlir-rewrite/CMakeFiles/mlir-rewrite.dir/mlir-rewrite.cpp.o.d -o tools/mlir-rewrite/CMakeFiles/mlir-rewrite.dir/mlir-rewrite.cpp.o -c /build/mlir-src-20.1.0/mlir/tools/mlir-rewrite/mlir-rewrite.cpp
mlir> In file included from /build/mlir-src-20.1.0/mlir/include/mlir/IR/UseDefLists.h:16,
mlir>                  from /build/mlir-src-20.1.0/mlir/include/mlir/IR/Value.h:17,
mlir>                  from /build/mlir-src-20.1.0/mlir/include/mlir/IR/BlockSupport.h:16,
mlir>                  from /build/mlir-src-20.1.0/mlir/include/mlir/IR/OperationSupport.h:18,
mlir>                  from /build/mlir-src-20.1.0/mlir/include/mlir/IR/AsmState.h:18,
mlir>                  from /build/mlir-src-20.1.0/mlir/include/mlir/AsmParser/AsmParser.h:16,
mlir>                  from /build/mlir-src-20.1.0/mlir/tools/mlir-rewrite/mlir-rewrite.cpp:13:
mlir> /build/mlir-src-20.1.0/mlir/include/mlir/IR/Location.h:145:10: fatal error: mlir/IR/BuiltinLocationAttributes.h.inc: No such file or directory
mlir>   145 | #include "mlir/IR/BuiltinLocationAttributes.h.inc"
mlir>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/pkgs/development/compilers/llvm/common/libc/default.nix
+++ b/pkgs/development/compilers/llvm/common/libc/default.nix
@@ -77,6 +77,7 @@ stdenv.mkDerivation (finalAttrs: {
   };
 
   meta = llvm_meta // {
+    broken = stdenv.hostPlatform.isDarwin;
     homepage = "https://libc.llvm.org/";
     description = "Standard C library for LLVM";
   };

--- a/pkgs/development/compilers/llvm/common/llvm/default.nix
+++ b/pkgs/development/compilers/llvm/common/llvm/default.nix
@@ -352,12 +352,19 @@ stdenv.mkDerivation (finalAttrs: {
     ] ++ [
       "-DLLVM_TABLEGEN=${buildLlvmTools.tblgen}/bin/llvm-tblgen"
     ];
+
+    triple =
+      if stdenv.hostPlatform.isDarwin && lib.versionAtLeast release_version "20" then
+        # JIT tests expect the triple to use Darwin arch's naming for CPU architectures.
+        "${stdenv.hostPlatform.darwinArch}-apple-${stdenv.hostPlatform.darwinPlatform}"
+      else
+        stdenv.hostPlatform.config;
   in flagsForLlvmConfig ++ [
     "-DLLVM_INSTALL_UTILS=ON"  # Needed by rustc
     "-DLLVM_BUILD_TESTS=${if finalAttrs.finalPackage.doCheck then "ON" else "OFF"}"
     "-DLLVM_ENABLE_FFI=ON"
-    "-DLLVM_HOST_TRIPLE=${stdenv.hostPlatform.config}"
-    "-DLLVM_DEFAULT_TARGET_TRIPLE=${stdenv.hostPlatform.config}"
+    "-DLLVM_HOST_TRIPLE=${triple}"
+    "-DLLVM_DEFAULT_TARGET_TRIPLE=${triple}"
     "-DLLVM_ENABLE_DUMP=ON"
     (lib.cmakeBool "LLVM_ENABLE_TERMINFO" enableTerminfo)
   ] ++ optionals (!finalAttrs.finalPackage.doCheck) [

--- a/pkgs/development/compilers/llvm/common/mlir/default.nix
+++ b/pkgs/development/compilers/llvm/common/mlir/default.nix
@@ -71,6 +71,8 @@ stdenv.mkDerivation rec {
   outputs = [ "out" "dev" ];
 
   meta = llvm_meta // {
+    # Very broken since the dependencies aren't propagating at all with tblgen through the CMake.
+    broken = lib.versionAtLeast release_version "20";
     homepage = "https://mlir.llvm.org/";
     description = "Multi-Level IR Compiler Framework";
     longDescription = ''

--- a/pkgs/development/compilers/llvm/default.nix
+++ b/pkgs/development/compilers/llvm/default.nix
@@ -30,7 +30,7 @@ let
     "17.0.6".officialRelease.sha256 = "sha256-8MEDLLhocshmxoEBRSKlJ/GzJ8nfuzQ8qn0X/vLA+ag=";
     "18.1.8".officialRelease.sha256 = "sha256-iiZKMRo/WxJaBXct9GdAcAT3cz9d9pnAcO1mmR6oPNE=";
     "19.1.7".officialRelease.sha256 = "sha256-cZAB5vZjeTsXt9QHbP5xluWNQnAHByHtHnAhVDV0E6I=";
-    "20.1.0-rc3".officialRelease.sha256 = "sha256-mLSBoyq24FD+khynC6sC5IkDFqizgAn07lR9+ZRXlV0=";
+    "20.1.1".officialRelease.sha256 = "sha256-hDFYi5wAUDQ8xfjvbGFi05KdawrN9lwrrSC8AcTpCEE=";
     "21.0.0-git".gitRelease = {
       rev = "4d928d5b5858de18f47f68c0d309dafd24cea875";
       rev-version = "21.0.0-unstable-2025-02-23";

--- a/pkgs/top-level/aliases.nix
+++ b/pkgs/top-level/aliases.nix
@@ -1680,7 +1680,7 @@ mapAliases {
   openbsdCross = openbsd; # Added 2024-09-06
 
   # LLVM packages for (integration) testing that should not be used inside Nixpkgs:
-  llvmPackages_latest = llvmPackages_19;
+  llvmPackages_latest = llvmPackages_20;
 
   /* If these are in the scope of all-packages.nix, they cause collisions
     between mixed versions of qt. See:

